[Issue] panic: interface conversion crash

panic: interface conversion: interface {} is string, not int

goroutine 12220628 [running]:
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).Match(0xc0000342c0, 0xc003e3ba40, 0xc00021c780, 0xc004a16880)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/pkg/protocols/http/operators.go:27 +0x276
github.com/projectdiscovery/nuclei/v2/pkg/operators.(*Operators).Execute(0xc0000342c0, 0xc003e3ba40, 0xc005575aa0, 0xc005575a90, 0xc00561cc08, 0xe)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/pkg/operators/operators.go:106 +0x591
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).executeRequest(0xc0000342c0, 0xc005cb6450, 0x11, 0xc0058fbd10, 0xc0058fbbc0, 0xc005575db8, 0x1, 0x0, 0x0)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/pkg/protocols/http/request.go:395 +0x156e
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).ExecuteWithResults(0xc0000342c0, 0xc005cb6450, 0x11, 0xc0058fbb90, 0xc0058fbbc0, 0xc0058fbc50, 0x1000, 0x7fa47c0790e8)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/pkg/protocols/http/request.go:211 +0x245
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0xc0000407e0, 0xc005cb6450, 0x11, 0xc005e514a0, 0xc001614f48, 0x407215)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/pkg/protocols/common/executer/executer.go:53 +0x14d
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).processTemplateWithList.func1.1(0xc0052c8000, 0xc0000b8340, 0xc00025e000, 0xc0052bc008, 0xc005cb6450, 0x11)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/internal/runner/processor.go:21 +0x9d
created by github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).processTemplateWithList.func1
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.3.0/internal/runner/processor.go:18 +0xfb

--- a/v2/pkg/protocols/http/operators.go
+++ b/v2/pkg/protocols/http/operators.go
@@ -24,7 +24,11 @@ func (r *Request) Match(data map[string]interface{}, matcher *matchers.Matcher) 
 		if !ok {
 			return false
 		}
-		return matcher.Result(matcher.MatchStatusCode(statusCode.(int)))
+		status, ok := statusCode.(int)
+		if !ok {
+			return false
+		}
+		return matcher.Result(matcher.MatchStatusCode(status))
 	case matchers.SizeMatcher:
 		return matcher.Result(matcher.MatchSize(len(item)))
 	case matchers.WordsMatcher:

